sw/device/tests/spi_host_irq_test.c

```diff
@@ -0,0 +1,181 @@
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/spi_device_testutils.h"
+#include "sw/device/lib/testing/spi_flash_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+static_assert(__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__,
+              "This test assumes the target platform is little endian.");
+
+OTTF_DEFINE_TEST_CONFIG();
+
+dif_spi_host_t spi_host;
+
+/**
+ * Declared volatile because it is referenced in the main program flow as well
+ * as the ISR.
+ */
+// Hold the test result.
+static volatile status_t test_result;
+// Used to sync the irs and the main thread.
+static volatile dif_spi_host_irq_t irq_fired;
+static dif_rv_plic_t plic;
+
+enum {
+  kHart = kTopEarlgreyPlicTargetIbex0,
+  kTimeoutMicros = 10000,
+};
+
+/**
+ * Provides external IRQ handling for this test.
+ *
+ * This function overrides the default OTTF external ISR.
+ *
+ * For each IRQ, it performs the following:
+ * 1. Claims the IRQ fired (finds PLIC IRQ index).
+ * 2. Checks that the index belongs to the expected peripheral.
+ * 3. Checks that the correct and the only IRQ from the expected peripheral
+ *    triggered.
+ * 4. Clears the IRQ at the peripheral.
+ * 5. Completes the IRQ service at PLIC.
+ */
+static status_t external_isr(void) {
+  dif_rv_plic_irq_id_t plic_irq_id;
+  // SPI_HOST_STATUS_DUMP(spi_host);
+  TRY(dif_rv_plic_irq_claim(&plic, kHart, &plic_irq_id));
+  LOG_INFO("%s: %d", __func__, plic_irq_id);
+
+  top_earlgrey_plic_peripheral_t peripheral = (top_earlgrey_plic_peripheral_t)
+      top_earlgrey_plic_interrupt_for_peripheral[plic_irq_id];
+  TRY_CHECK(peripheral == kTopEarlgreyPlicPeripheralSpiHost0,
+            "IRQ from incorrect peripheral: exp = %d(spi_host0), found = %d",
+            kTopEarlgreyPlicPeripheralSpiHost0, peripheral);
+
+  irq_fired = (dif_spi_host_irq_t)(plic_irq_id -
+                                   (dif_rv_plic_irq_id_t)
+                                       kTopEarlgreyPlicIrqIdSpiHost0Error);
+
+  TRY(dif_spi_host_irq_acknowledge(&spi_host, irq_fired));
+
+  // Complete the IRQ at PLIC.
+  TRY(dif_rv_plic_irq_complete(&plic, kHart, plic_irq_id));
+  return OK_STATUS();
+}
+
+void ottf_external_isr(void) { test_result = external_isr(); }
+
+static status_t ready_event_irq(void) {
+  uint8_t data[256];
+  memset(data, 0xA5, sizeof(data));
+  dif_spi_host_status_t status;
+
+  irq_fired = UINT32_MAX;
+  TRY(dif_spi_host_event_set_enabled(&spi_host, kDifSpiHostEvtReady, true));
+
+  TRY(dif_spi_host_get_status(&spi_host, &status));
+  TRY_CHECK(status.ready);
+  TRY_CHECK(!status.active);
+
+  // Overwhelm the cmd fifo to make the `STATUS.ready` goes low.
+  TRY(dif_spi_host_fifo_write(&spi_host, data, sizeof(data)));
+  for (int i = 0; i < 5; ++i) {
```

Comment:
```suggestion
  for (size_t i = 0; i < 5; ++i) {
```
Also, maybe you should pick an even multiple of sizeof(data), i.e., 4 or 8? instead of 5, so all the data gets sent? You could also check it all gets sent below too.

Reply:
Done

Reply:
doesn't look like this was fixed?